docs/proposals/default-service-plans.md

```diff
@@ -0,0 +1,544 @@
+
+## Abstract
+
+Let's wholesale rip off how default storage classes for volumes work and apply
+it to Service Catalog! 😎 This allows us to specify a dependency on a type of
+service without requiring upfront knowledge of the underlying broker.
+
+> "My application requires a mysql database."
+
+> "I just want to try out this application using the recommended defaults."
+
+> "I need to encourage everyone to use the cheapest plans possible in our
+> integration environment by making a small database the default."
+
+## Summary
+
+1.  Add default provisioning and binding parameters to Service Classes.
+1.  Add a "Service Type" field to classes and plans, e.g. mysql or redis.
+1.  Add a is-default-plan annotation to Service Plans to flag it as the default
+    plan for a particular service type.
+1.  Service Instances and Bindings fall back to the defaults set on the class  
+    when parameters are not provided.
+1.  Services can be provisioned using only the Service Type, e.g. mysql, and the
+    default plan and parameters are used.
+
+## Motivation
+
+Consider a Helm chart author who is interested in updating their chart to be
+"Service Catalog Ready"\*. Their chart requires a MySQL database and they'd like
+to help their users by setting up the database in their chart, just like the
+upstream Helm charts do today.
+
+So they add a ServiceInstance to their chart, and when they get to the class and
+plan fields in the spec, they hit a wall. The author doesn't know the
+capabilities of the cluster on which their chart will be deployed and aren't in
+a position to provide the class and plan. Neither do they know the secret keys,
+as it's broker specific. The author is forced to add required arguments to their
+chart, passing the buck on figuring out the right class, plan, instance
+parameters and secret keys to the person deploying the chart.
+
+Now consider a user who wants to get up and running quickly, so naturally they
+chose to use a "Service Catalog Ready"* Helm chart. They don't know or care
+about the details, and want to get the application deployed with sane defaults
+so that they can try it out. Before they can even install the application they
+need to:
+
+* Realize that they need to specify a class and plan.
+* Figure out what a broker is and which one is installed on their cluster.
+* Hunt down a list of suitable classes and plans.
+* Decide which class and plan is right for a trying out the application.
+* Fill in the class and plan, install the chart only to later learn that there
+    is a set of "magic" parameters that they should have specified when
+    provisioning the instance.
+* Scour the internet looking for undocumented, totally optional but it won't
+    work without them, parameters for their broker because brokers aren't
+    required to provide a schema.
+* Blindly copy/paste some parameters they found in a gist somewhere.
+* Realize that the broker is returning MYSQL_PASSWORD, while the app expects
```

Comment:
This could be https://github.com/kubernetes-incubator/service-catalog/issues/1311

Reply:
I didn't know about this when I wrote it but I think it's a perfect fit. That feature stays as-is, and this proposal enables people to define default mappings so that each binding doesn't have to do the heavy lifting of defining the mapping each time.